src/schema/v2/me/conversation/conversationEventConnection.ts

```diff
@@ -0,0 +1,201 @@
+  },
+})
+
+const ConversationEventType = new GraphQLUnionType({
+  name: "ConversationEvent",
+  types: [
+    MessageType,
+    ConversationOfferReceivedEventType,
+    ConversationOrderStateChangedType,
+  ],
+  resolveType: (value) => {
+    // debugger
+    const { __typename } = value
+    switch (__typename) {
+      case "CommerceOrderStateChangedEvent":
+        return "ConversationOrderStateChanged"
+      case "CommerceOfferSubmittedEvent":
+        return "ConversationOfferSubmitted"
+      default:
+        return "Message"
+    }
+  },
+})
+
+const lastMessageId = (conversation) => {
+  return get(conversation, "_embedded.last_message.id")
+}
+
+export const conversationEventConnection: GraphQLFieldConfig<
+  any,
+  ResolverContext
+> = {
+  description: "Messages and other events in a conversation.",
+  type: connectionWithCursorInfo({
+    nodeType: ConversationEventType,
+  }).connectionType,
+  args: pageable({
+    sort: {
+      type: new GraphQLEnumType({
+        name: "sortMessages",
+        values: {
+          DESC: { value: "desc" },
+          ASC: { value: "asc" },
+        },
+      }),
+    },
+  }),
+  resolve: (
+    { id, from_email, initial_message, from_name, _embedded },
+    options,
+    { conversationMessagesLoader }: { conversationMessagesLoader?: any }
+  ) => {
+    if (!conversationMessagesLoader) return null
+    const optionKeys = Object.keys(options)
+    if (optionKeys.includes("last") && !optionKeys.includes("before")) {
+      options.before = `${lastMessageId({ _embedded })}`
+    }
+    const { page, size, offset } = convertConnectionArgsToGravityArgs(options)
+    return conversationMessagesLoader({
+      page,
+      size,
+      conversation_id: id,
+      "expand[]": "deliveries",
+      sort: options.sort || "asc",
+    }).then(({ total_count, message_details }) => {
+      // Inject the convesation initiator's email into each message payload
+      // so we can tell if the user sent a particular message.
+      // Also inject the conversation id, since we need it in some message
+      // resolvers (invoices).
+      /* eslint-disable no-param-reassign */
+      const messages = message_details.map((message) => {
+        return merge(message, {
+          conversation_initial_message: initial_message,
+          conversation_from_name: from_name,
+          conversation_from_address: from_email,
+          conversation_id: id,
+        })
+      })
+      const orderEvents = orderEventStub.map((event, index) => {
+        return { ...event, id: `event-${index}` }
+      })
+      const conversationEvents = sortBy(
+        [...messages, ...orderEvents],
+        (event) => new Date(event.createdAt!)
+      )
+      debugger
+      return {
+        ...connectionFromArraySlice(conversationEvents, options, {
+          arrayLength: total_count + orderEvents.length,
+          sliceStart: offset,
+        }),
+        totalCount: total_count + orderEvents.length,
```

Comment:
the messages give us a total count and we currently assume the exchange query is giving us all order events, so we just add them together. I don't know if what i have above is the best way to do it or if there are other properties that need to be tweaked aside from `totalCount`.